vendor/k8s.io/kubernetes/staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -184,8 +187,34 @@ func (c *AvailableConditionController) sync(key string) error {
 			return err
 		}
 	}
-
-	// TODO actually try to hit the discovery endpoint
+	// actually try to hit the discovery endpoint
+	if apiService.Spec.Service != nil {
+		destinationHost := getDestinationHost(apiService, c.serviceLister)
+		discoveryURL := "https://" + destinationHost + "/apis"
+		// construct an http client that will ignore TLS verification (if someone owns the network and messes with your status
+		// that's not so bad) and sets a very short timeout.
+		httpClient := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+		}
+		req, err := http.NewRequest("GET", discoveryURL, nil)
+		if err != nil {
+			return err
+		}
+		ctx, _ := context.WithTimeout(req.Context(), 5*time.Second)
+		req.WithContext(ctx)
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			availableCondition.Status = apiregistration.ConditionFalse
```

Comment:
> what's the retry behaviour if this fails?

It will retry
 1. when the apiservice resyncs
 2. when the service changes
 3. when the endpoints change

We can add a separate timer if you like, but those ought not take too long.